### ISSUE TITLE
object variable errors in function parameter

### ISSUE THREAD

**Issue by FortyTwoFortyTwo** [alliedmodders/sourcepawn#458]
`object` as a variable name is disallowed and reports an error. This code reports an error fine
```js
public void ObjectVariable(int object) {}
```

However this code with extra param silently fails spcomp with no error report, as i expected any error report from this
```js
public void ObjectVariable(int object, int blah) {}
```

**Comment by dvander** [comment]
This is a reserved keyword.

**Comment by dvander** [comment]
Whoops - I misread, will investigate.

**Comment by kjsmita6** [comment]
This issue is almost 2 years old but I am still running into this using 1.10.0-6524. The compiler crashes and does not finish. Using 1.11 compiler I am able to get these errors with the code:
```
public void CategoryHandler(TopMenu topmenu, TopMenuAction action, TopMenuObject object, int param, char[] buffer, int maxlen) {
    if (action == TopMenuAction_DisplayTitle) {
        Format(buffer, maxlen, "Commands:");
    } else if (action == TopMenuAction_DisplayOption) {
        Format(buffer, maxlen, "Commands");
    }
}

public void AdminMenu_RA(TopMenu topmenu, TopMenuAction action, TopMenuObject object, int param, char[] buffer, int maxlen) {
    if (action == TopMenuAction_DisplayOption) {
        Format(buffer, maxlen, "Command Name");
    } else if (action == TopMenuAction_SelectOption) {
        FakeClientCommand(param, "say /command");
    }
}
```

Errors on the first line:
```
filename.sp(76) : error 001: expected token: ")", but found "object"
filename.sp(76) : error 001: expected token: "{", but found "object"
filename.sp(76) : error 003: declaration of a local variable must appear in a compound block
```

**Comment by asherkin** [comment]
This was fixed in the branch targeting 1.11, so the behaviour you're seeing is as expected.
